packages/core/src/sync-realtime/index.ts

```diff
@@ -497,6 +497,13 @@ export const createRealtimeSync = (
       requiredTransactions.has(hash),
     );
 
+    // Validate that filtered logs/callTraces point to valid transaction in the block
+    if (transactions.length !== requiredTransactions.size) {
+      throw new Error(
+        `Received callTrace/log in the block referencing transaction that is not in current head block '${block.hash}'`,
+      );
```

Comment:
Not blocking, but I'd advocate for a more specific error message which includes the suspicious transaction hashes. That would make it much easier for us to submit bug reports to chains / RPC providers.

Also, what if the lengths are different because we have an _extra_ required transaction? That scenario would suggest a Ponder bug rather than a faulty RPC.

Reply:
1) With regards to error message, do you think it would be better to include the list of suspicious tx hashes in the message? 
2) We would expect requiredTransactions size to be larger than transactions length only in the case if some logs or callTraces that is matched with filter are pointing to more transactions, than there are transactions filtered by those logs/traces. 
`const transactions = block.transactions.filter(({ hash }) =>
      requiredTransactions.has(hash),
    );`

Reply:
1. yes
2. I think the transaction validation should be more precise. Something like:
```ts
const blockTransactionsHashes = new Set(
  block.transactions.map((t) => t.hash),
);
for (const hash of Array.from(requiredTransactions)) {
  if (blockTransactionsHashes.has(hash) === false) {
    // throw error, include `hash`
  }
}
```

I also think it would be nice to have something that compares `blockHash` in `logs` and `callTraces` to `block.hash`